src/components/Basket.js

```diff
@@ -0,0 +1,67 @@
+import React from "react";
+
+
+class Basket extends React.Component {
+  constructor() {
+    super();
+     this.basketAdd = this.basketAdd.bind(this);
+    this.basketRemove = this.basketRemove.bind(this);
+    this.state = {order:{}}
+   
+  }
+
+  basketAdd(foodItem){
+    const newBasket = Object.assign({}, this.state.basket);
+    if (this.state.basket[foodItem.id]) {
+      newBasket[foodItem.id] += 1
+    } else {
+      newBasket[foodItem.id] = 1
+    }
+
+    this.setState({ basket: newBasket }, () => {
+      console.log(this.state);
+    })    
+  }
+
+  basketRemove(foodItem){
+    const newBasket = Object.assign({}, this.props.basket);
+    if (this.props.basket[foodItem.id]) {
+      newBasket[foodItem.id] -= 1
+    } else {
+      newBasket[foodItem.id] = 0
```

Comment:
it might make sense to use delete the current item from basket here rather than set quantity to 0.